### PR TITLE
Create `NonZero` trait with primitive associated type

### PR DESCRIPTION
**Pull request by nvzqz**
This enables getting the primitive integer type for a nonzero type through a type alias.

r? @yaahc

### CODE REVIEW

File: library/core/src/num/nonzero.rs [hunk 1]
```diff
@@ -6,7 +6,32 @@ use crate::str::FromStr;
...
+    /// state its primitive type.
+    ///
+    /// ```
+    /// use std::num::{NonZero, NonZeroU32};
+    ///
+    /// type NonZeroInt = NonZeroU32;
+    ///
+    /// let n: <NonZeroInt as NonZero>::Scalar = 1;
+    /// ```
+    type Scalar;
```
Comment (nvzqz) [inline library/core/src/num/nonzero.rs:33]: I'm open to suggestions for a better name.

**Comment by yaahc** [comment]
@nvzqz can you add more detail to the top level comment explaining how you plan to use this API? I'm a little confused by the description you gave, is the NonZero type the one being aliased or are you creating an alias using the type exposed via this trait? Ideally I'd love to see a link to a branch on a crate where you intend to use this API testing out the API and demonstrating the intended usage.

**Comment by JohnCSimon** [comment]
@nvzqz 
Ping from triage: I'm closing this due to inactivity, Please reopen when you are ready to continue with this. 
Note: if you do please open the PR BEFORE you force push to it, else you won't be able to reopen.
Thanks for your contribution.

@rustbot label: +S-inactive

**Pull request closed without merging**